automata/fa/dfa.py

```diff
@@ -652,119 +569,146 @@ def complement(self, *, retain_names: bool = False, minify: bool = True) -> DFA:
             allow_partial=self.allow_partial,
         )
 
-    # visited_set, product_transitions, product_initial_state_name, final_states
-    DFADataTupleT = Tuple[Set[DFAStateT], DFATransitionsT, DFAStateT, Set[DFAStateT]]
+    @staticmethod
+    def _bfs_edges(
+        initial_state: DFAStateT,
+        expand_state_fn: ExpandStateFn,
+    ) -> Iterable[Tuple[DFAStateT, DFASymbolT, DFAStateT]]:
```

Comment:
Same as other comment, I think it's better to use a `Generator` annotation.

Reply:
@eliotwrobson Also noting that I agree `Generator` should be used as the return type.

Reply:
@EduardoGoulart1 Also noting that I agree Generator should be used as the return type.

Reply:
Committed this tweak myself, so marking this thread as resolved.